software/apps/synth/main.c

```diff
@@ -13,27 +14,89 @@
 #include "runtime.h"
 #include "synchronization.h"
 
-#define N 16
+// Macros
+#define UNROLL (4)
+#define N (NUM_CORES * UNROLL)
+#define TARGET_MEMPOOL (1)
+#define TARGET_TERAPOOL (0)
 
-int32_t block_a[N] __attribute__((section(".l1_prio")));
-int volatile error __attribute__((section(".l1")));
+// Globals
+int32_t block_a[N] __attribute__((aligned(NUM_CORES * 4), section(".l1")));
```

Comment:
Where is this four coming from? Do you want to align to the L1 start so all accesses are local?